src/utils/stream.ts

```diff
@@ -0,0 +1,35 @@
+export class StreamingApi {
+  private writer: WritableStreamDefaultWriter<Uint8Array>
+  private encoder: TextEncoder
+  buffer: string[] = []
+
+  constructor(writable: WritableStream<Uint8Array>) {
+    this.writer = writable.getWriter()
+    this.encoder = new TextEncoder()
+  }
+
+  write(str: string): StreamingApi {
+    this.buffer.push(str)
```

Comment:
Does it make sense to buffer here into memory or just write directly to the stream? If the latter, I don't think you'd even need `.flush()` at all?

Reply:
Hi @geelen, thanks for your review.
I found a `flush` function in @geelen 's reference implementation, so I implemented this to create a buffer until then so that the stream is written at the timing when the `flush` actually occurs.
Surely, I think it is unnecessary because the user can buffer the stream before writing to it.
Would it be OK to send it when `write` or `writeln` is called?

Reply:
Yeah, I was testing something at the time with Cloudflare Workers and found that it was only streaming in 4096 byte chunks, so I made flush send a bunch of bytes of whitespace to force it to send 🤫 

I think I was doing something wrong, though, since when I tried again, chunks were streamed much more rapidly so I think it's fine to send them as soon as `write` is called. A user can always buffer on their own side before sending it...